core/src/tile/tileManager.cpp

```diff
@@ -43,18 +43,14 @@ void TileManager::setTileSources(const std::vector<std::shared_ptr<TileSource>>&
 
     m_tileCache->clear();
 
-    // remove sources that are not in new scene - there must be a better way..
+    // Remove all (non-client datasources) sources and respective tileSets not present in the
+    // new scene
     auto it = std::remove_if(
         m_tileSets.begin(), m_tileSets.end(),
         [&](auto& tileSet) {
             if (!tileSet.clientTileSource) {
-                auto sIt = std::find_if(_sources.begin(), _sources.end(),
-                                        [&](auto& source){ return source->equals(*tileSet.source); });
-
-                if (sIt == _sources.end() || !(*sIt)->generateGeometry()) {
-                    LOGD("remove source %s", tileSet.source->name().c_str());
-                    return true;
-                }
+                LOGD("remove source %s", tileSet.source->name().c_str());
```

Comment:
@hjanetzek This is what you were referring to in https://github.com/tangrams/tangram-es/pull/1498#discussion_r117051629.

Also though it is good to have the `TileSource::equals` functionality, with this change it is not used now. Does it make sense to remove it, I am fine either way, but probably can leave it.